packages/block-editor/src/components/rich-text/use-input-rules.js

```diff
@@ -42,6 +42,26 @@ function findSelection( blocks ) {
 	return [];
 }
 
+function convertSpaces( value ) {
```

Comment:
this function could use some explanation, and possible a clearer name like `replacePrecedingSpaces`

just a single sentence explaining its purpose could help quite a bit.